pkg/operator/targetconfigcontroller/targetconfigcontroller_test.go

```diff
@@ -162,3 +162,18 @@ func TestTargetConfigController(t *testing.T) {
 		})
 	}
 }
+
+func TestBackVarsToArgs(t *testing.T) {
+	input := map[string]string{
+		"data-dir":   "/var/lib/etcd",
+		"config-dir": "/etc/kubernetes",
+		"backup-dir": "/var/backup/etcd",
+	}
+
+	expected := "- --data-dir=/var/lib/etcd\n- --config-dir=/etc/kubernetes\n- --backup-dir=/var/backup/etcd\n"
+
+	act := backVarsToArgs(input)
+	// map order is non-deterministic
+	// use contains instead of equal
+	require.Contains(t, expected, act)
```

Comment:
`require.Equal` takes care of that using reflect.DeepEquals, not sure why you think this matters

Reply:
it fails, i used it first, ok will try again :D

Reply:
then it's not the same string :rofl: Please avoid this string map stuff, let's use the proper CRD types...

Reply:
it failed again 
```
=== RUN   TestBackupConfig_ToArgs
    backupvars_test.go:26: 
        	Error Trace:	/Users/melbeher/go/src/openshift/cluster-etcd-operator/pkg/backuphelpers/backupvars_test.go:26
        	Error:      	Not equal: 
        	            	expected: "- --data-dir=/var/lib/etcd\n- --config-dir=/etc/kubernetes\n- --backup-dir=/var/backup/etcd"
        	            	actual  : "- --backup-dir=/var/backup/etcd\n- --data-dir=/var/lib/etcd\n- --config-dir=/etc/kubernetes"
        	            	
        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -1,3 +1,3 @@
        	            	+- --backup-dir=/var/backup/etcd
        	            	 - --data-dir=/var/lib/etcd
        	            	 - --config-dir=/etc/kubernetes
        	            	-- --backup-dir=/var/backup/etcd
        	Test:       	TestBackupConfig_ToArgs
--- FAIL: TestBackupConfig_ToArgs (0.00s)
```

the order of the values within the map generates diff string

Reply:
done 👍🏽